File: test_javaPrj_01/src/programmers/lv1/추억점수.java
package programmers.lv1;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class 추억점수 {

    private 추억점수(){}

    public void Solution() throws Exception{
        
        String[] name = {"may", "kein", "kain", "radi"};
        int[] yearning = {5, 10, 1, 3};
        String[][] photo = {{"may", "kein", "kain", "radi"},{"may", "kein", "brin", "deny"}, {"kon", "kain", "may", "coni"}};

        Map<String,Integer> peopleHash = new HashMap<>();
        for(int i = 0; i < name.length; i++){
            peopleHash.put(name[i], yearning[i]);
        }
        
        int[] scoreArr = new int[photo.length];
        for(int i = 0; i < photo.length; i++){

            int totalScore = 0;
            for(int j = 0; j < photo[i].length; j++){
                if(peopleHash.get(photo[i][j]) != null){
                    totalScore += peopleHash.get(photo[i][j]);
                }
            }

            scoreArr[i] = totalScore;
        }
        
        System.out.println(Arrays.toString(scoreArr));
    }


    public static void main(String[] args) {
        
        try{
            new 추억점수().Solution();
        }catch(Exception e){
            e.printStackTrace();
        }


    }
    
}

File: test_javaPrj_01/src/programmers/lv1/추억점수.py


class 추억점수:

    def __init__(self, name:list[str], yearning:list[int], photo:list[list[str]]) -> None:

    
        o = [ sum( yearning[name.index(j)] for j in p if j in name ) for p in photo]

        testlist = [1,2,3,4]

        zipFucn = zip(name,yearning,testlist)
        dictionary = dict(zip(name,yearning))

        print(list(zipFucn))
        print(dictionary)
        print( o )

        # zip함수
        # zip함수 + list,dict,tuple의 조합

        peopleHash = dict(zip(name, yearning))
        o2 = [ sum( peopleHash.get(person) if peopleHash.get(person) else 0 for person in aCopy ) for aCopy in photo ]

        print(o2)
        pass


추억점수(["may", "kein", "kain", "radi"], [5, 10, 1, 3], [["may", "kein", "kain", "radi"],["may", "kein", "brin", "deny"], ["kon", "kain", "may", "coni"]])
# 추억점수(["may", "kein", "kain", "radi"], [5, 10, 1, 3], [["may", "kein", "kain", "radi"]])